File: python/messages_dapp/dapp.py
from pwrpy.pwrwallet import Wallet
from sync_messages import sync
from dotenv import load_dotenv
import json
import threading
import os
load_dotenv()

# Setting up your wallet in the SDK
seed_phrase = os.getenv("SEED_PHRASE")
wallet = Wallet.new(seed_phrase)
vida_id = 1234

def main():
    threading.Thread(target=sync, daemon=True).start()

    while True:
        message = input("")
        obj = {"message": message}
        data = json.dumps(obj).encode('utf-8')
        
        # Send the VIDA data
        response = wallet.send_vida_data(vida_id, data)
        if response.success==False:
            print('FAILED!')
main()

File: go/messagesDapp/dapp.go
package messagesDapp

import (
    "fmt"
    "bufio"
    "os"
    "encoding/json"
    "github.com/pwrlabs/pwrgo/wallet"
    "github.com/joho/godotenv"
)

func main() {
    // Setting up your wallet in the SDK
    godotenv.Load()
    seedPhrase := os.Getenv("SEED_PHRASE")
    wallet, err := wallet.New(seedPhrase)
    if err != nil {
        fmt.Println("Error getting the wallet:", err)
        return
    }
    vidaId := 1234 // Replace with your VIDA's ID

    go Sync()

    reader := bufio.NewReader(os.Stdin)

    for {
        fmt.Print("Enter your message: ")
        message, _ := reader.ReadString('\n')
        message = message[:len(message)-1]

        object := map[string]string{"message": message}
        jsonData, err := json.Marshal(object)
        if err != nil {
            fmt.Println("Failed to encode message:", err)
            continue
        }
        feePerByte := wallet.GetRpc().GetFeePerByte()

        // Send the VIDA data
        tx := wallet.SendVidaData(vidaId, jsonData, feePerByte)
        if tx.Success {
            fmt.Printf("Transaction Hash: %s\n", tx.Hash)
        } else {
            fmt.Println("Error:", tx.Error)
        }
    }
}